manim/utils/config_utils.py

```diff
@@ -0,0 +1,404 @@
+        "the rendering at the second value",
+    )
+
+    # Specify the manim.cfg file
+    parser.add_argument(
+        "--config_file", help="Specify the configuration file",
+    )
+
+    return parser.parse_args(arg_list)
+
+
+def _init_dirs(config):
+    # Make sure all folders exist
+    for folder in [
+        config["media_dir"],
+        config["video_dir"],
+        config["tex_dir"],
+        config["text_dir"],
+    ]:
+        if not os.path.exists(folder):
+            os.makedirs(folder)
+
+
+def _from_command_line():
+    """Determine if manim was called from the command line."""
+    # Manim can be called from the command line in three different
+    # ways.  The first two involve using the manim or manimcm commands
+    prog = os.path.split(sys.argv[0])[-1]
+    from_cli_command = prog in ["manim", "manimcm"]
+
+    # The third way involves using `python -m manim ...`.  In this
+    # case, the CLI arguments passed to manim do not include 'manim',
+    # 'manimcm', or even 'python'.  However, the -m flag will always
+    # be the first argument.
+    from_python_m = sys.argv[0] == "-m"
+
+    return from_cli_command or from_python_m
+
+
+def _paths_config_file():
+    library_wide = os.path.abspath(
+        os.path.join(os.path.dirname(__file__), "..", "default.cfg")
+    )
+    if sys.platform.startswith("linux"):
+        # This is for linux users
+        user_wide = os.path.expanduser(
+            os.path.join("~", ".config", "Manim", "manim.cfg")
+        )
+    elif sys.platform.startswith("darwin"):
+        # This is for MacOS users
+        user_wide = os.path.expanduser(
+            os.path.join("~", "Library", "Application Support", "Manim", "manim.cfg")
+        )
+    elif sys.platform.startswith("win32"):
+        # This is for Windows users
+        user_wide = os.path.expanduser(
+            os.path.join("~", "AppData", "Roaming", "Manim", "manim.cfg")
+        )
+    else:
+        # This is for users whose OS is unknown
+        user_wide = os.path.expanduser(
+            os.path.join("~", ".config", "Manim", "manim.cfg")
+        )
+    current_working_directory = os.path.abspath("manim.cfg")
+    return [library_wide, user_wide, current_working_directory]
+
+
+def _run_config():
+    # Config files to be parsed, in ascending priority
+    config_files = _paths_config_file()
+    if _from_command_line():
+        args = _parse_cli(sys.argv[1:])
+        if args.config_file is not None:
+            if os.path.exists(args.config_file):
+                config_files.append(args.config_file)
+            else:
+                raise FileNotFoundError(f"Config file {args.config_file} doesn't exist")
+        else:
+            script_directory_file_config = os.path.join(
+                os.path.dirname(args.file), "manim.cfg"
+            )
+            if os.path.exists(script_directory_file_config):
+                config_files.append(script_directory_file_config)
```

Comment:
OK, here comes a new question. Which `manim.cfg` should be taken at first? The one in the current working directory or the one the script is located? See below for an example.
I could do if I am in CWD and there is a `scene.py` I can do
```sh
manim scene.py
```
Now there would no problem reading the `manim.cfg` file there. But if I do 
```sh
manim /path/to/scene.py
```
Now, there is a dilemma of which cfg file to be used. The one in CWD or the one in which lives in the folder where `scene.py` is there. Pleace do help me with fixing this the I would be resolve this comment. @leotrs

Reply:
The python interpreter considers the place from which you executed the `python` command to be it's current working directory, even if you are running a file which is somewhere else.

In keeping with this tradition, I would suggest using the CWD's `.cfg` file, and not the one next to the script holding your animation.

Reply:
I strongly think that it should be the one that is next to the scene script. Otherwise, running the same command on the same script could generate different videos. I believe @eulertour has said something similar before.